src/main/java/seedu/address/logic/parser/AttendanceMarkingCommandParser.java

```diff
@@ -0,0 +1,73 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TELEGRAM;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AttendanceMarkingCommand;
+import seedu.address.logic.commands.MarkAttendanceCommand;
+import seedu.address.logic.commands.UnmarkAttendanceCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.attendance.Attendance;
+import seedu.address.model.person.Telegram;
+
+/**
+ * Parses input arguments and creates a new MarkAttendanceCommand or UnmarkAttendance object based on user input
+ */
+public class AttendanceMarkingCommandParser implements Parser<AttendanceMarkingCommand> {
+    private String markType;
+    public AttendanceMarkingCommandParser(String markType) {
+        this.markType = markType;
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the MarkAttendanceCommand or
+     * UnmarkAttendanceCommand based on {@code markType}
+     * and returns a MarkAttendanceCommand or UnmarkAttendanceCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AttendanceMarkingCommand parse(String args) throws ParseException {
```

Comment:
i think the parser will only get the last telegram and ignore the rest. so the user cannot mark multiple users currently.

Reply:
Currently the command format is aligned with other command that accept multiple value of the same type, eg. find command can allow search for multiple names by `n/name1 n/name2`, the current command to accept multiple telegrams is `t/telegram1 t/telegram2`, which could successfully detect multiple telegrams, and I think it would be better to follow command conventions instead of creating a new kind of command format.

Reply:
Oh my bad. it didn't work for me previously because one of my telehandle was invalid. I tested again and it works now 👍. Maybe for the next iteration can include feedback to the user so that they know which persons were successfully marked.

Reply:
I just changed the success message but I found this branch is already merged, so I created another PR, can merge it if necessary (it's just some tiny changes)